src/main/java/com/patterns/flyweight/VehicleFactory.java

```diff
@@ -0,0 +1,23 @@
+package com.patterns.flyweight;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class VehicleFactory {
+    private static final Map<VehicleType, Vehicle> VEHICLES = new HashMap<>();
+
+    public Vehicle getVehicle(VehicleType kindOfVehicle) {
+        Vehicle vehicle = VEHICLES.get(kindOfVehicle);
+        if (vehicle == null) {
```

Comment:
It is better to use switch-case here.
As each element of enum is singlton. So you could use == instead equals